### ISSUE TITLE
Enrichment Analysis doesn't work

### ISSUE THREAD

**Issue by FelixThaler** [hfang-bristol/XGR#8]
I'm trying to do Gene Enrichment analysis using the following code:

library(XGR)

RData.location <- "http://galahad.well.ox.ac.uk/bigdata"

res <- xRDataLoader(RData.customised='org.Hs.eg', RData.location=RData.location)

background <- res$Symbol

data <- as.character(Dupi_16w_vs_1y[ ,2])     # Dupi_16w_vs_1y is my dataframe containing
                                                                        # differentially expressed genes

xDefineOntology(ontology = 'KEGG')     # doesn't work, i get the same error message as shown later

Enriched_16w_vs_1y <- xEnricherGenes(data=data, background=background, ontology="KEGG", 
                                     ontology.algorithm="none", RData.location=RData.location)

at this point the console prints:

Start at 2020-02-19 16:12:15

Load the ontology KEGG and its gene annotations (2020-02-19 16:12:15) ...
Start at 2020-02-19 16:12:15

Attempt to download from https://github.com/hfang-bristol/RDataCentre/blob/master/Portal/org.Hs.egKEGG.RData?raw=true (at 2020-02-19 16:12:15)
Attempt to download from http://galahad.well.ox.ac.uk/bigdata/org.Hs.egKEGG.RData (at 2020-02-19 16:12:15)
Attempt to download from http://galahad.well.ox.ac.uk/bigdata/org.Hs.egKEGG.RData (at 2020-02-19 16:12:15)
'org.Hs.egKEGG' CANNOT be loaded (at 2020-02-19 16:12:15)

End at 2020-02-19 16:12:15
Runtime in total is: 0 secs

Error in data.frame(from = "root", to = nodes$name) : 
  arguments imply differing number of rows: 1, 0

does anyone know where i made the mistake(s)? 
thanks

**Comment by hfang-bristol** [comment]
Replace “KEGG” with “MsigdbC2KEGG”.

Just curious why you specify “KEGG“; there is no documentation suggesting to do so.

**Comment by FelixThaler** [comment]
> Replace “KEGG” with “MsigdbC2KEGG”.
> 
> Just curious why you specify “KEGG“; there is no documentation suggesting to do so.


Thanks a lot!

after fixing my mistake as you suggested the Error message dissappeared.
Unfortunately, when i try to proceed as shown in http://galahad.well.ox.ac.uk/XGR/XGR_differential_genes.html , my code doesn't seem to work. 

I used the following code. Could you tell me where i made the mistake? 
Thank you so much for your help!

> Enriched_16w_vs_1y <- xEnricherGenes(data=data, background=background, ontology="MsigdbC2KEGG", 
+                                      ontology.algorithm="none", RData.location=RData.location)
Start at 2020-02-19 20:50:01

Load the ontology MsigdbC2KEGG and its gene annotations (2020-02-19 20:50:01) ...
Start at 2020-02-19 20:50:01

'org.Hs.egMsigdbC2KEGG' (from http://galahad.well.ox.ac.uk/bigdata/org.Hs.egMsigdbC2KEGG.RData) has been loaded into the working environment (at 2020-02-19 20:50:02)

End at 2020-02-19 20:50:02
Runtime in total is: 1 secs

Do gene mapping from Symbols to EntrezIDs (2020-02-19 20:50:02) ...
Start at 2020-02-19 20:50:02

'org.Hs.eg' (from http://galahad.well.ox.ac.uk/bigdata/org.Hs.eg.RData) has been loaded into the working environment (at 2020-02-19 20:50:05)

End at 2020-02-19 20:50:05
Runtime in total is: 3 secs

human organism (2020-02-19 20:50:05)
Among 1 symbols of input data, there are 0 mappable via official gene symbols but 1 left unmappable

#######################################################
'xEnricher' is being called (2020-02-19 20:50:05):
#######################################################
First, generate a subgraph induced (via 'all_paths' mode) by the annotation data (2020-02-19 20:50:05) ...
Next, prepare enrichment analysis (2020-02-19 20:50:06) ...
#######################################################
'xEnricher' has been finished (2020-02-19 20:50:06)!
#######################################################


End at 2020-02-19 20:50:06
Runtime in total (xEnricherGenes): 5 secs

> xEnrichViewer(Enriched_16w_vs_1y, 10)
NULL
> xEnrichBarplot(Enriched_16w_vs_1y, top_num="auto", displayBy="fdr")
NULL

**Comment by hfang-bristol** [comment]
The running is successful (Enriched_16w_vs_1y) but Null returned because only one gene (official symbol) found from what you input. Please check your provided gene list (Official Symbols).

**Comment by FelixThaler** [comment]
Alright, thanks!
Should i convert to EntrezGeneID by using the xGeneID2Symbol function? My current data consists of gene symbols

**Comment by hfang-bristol** [comment]
Yes. You can provide GeneID then converted into official symbols using xGeneID2Symbol.

**Comment by FelixThaler** [comment]
Ok thanks. I tried to do so by using the following code:

`neue_daten = xGeneID2Symbol(daten, org = c("human"), details = T, # converting to EntrezGeneID
               verbose = T, RData.location = "http://galahad.well.ox.ac.uk/bigdata")`

but unfortunataley i got the Error message:

`1 GeneID of input data (2020-02-22 12:09:21)
Start at 2020-02-22 12:09:21

'org.Hs.eg' (from http://galahad.well.ox.ac.uk/bigdata/org.Hs.eg.RData) has been loaded into the working environment (at 2020-02-22 12:09:22)

End at 2020-02-22 12:09:22
Runtime in total is: 1 secs

human organism (2020-02-22 12:09:22)
0 mappable but 1 left unmappable (2020-02-22 12:09:22)
Warning message:
In xGeneID2Symbol(daten, org = c("human"), details = T, verbose = T,  :
  NAs introduced by coercion`

Changing the 'details' or the 'verbose' arguments doesn't help either. To check if my gene list is valid, i tried the enrichment analysis using the XGR web version (http://galahad.well.ox.ac.uk:3030/enricher/genes) and it worked. 

Could you please tell me if i did something wrong?
Thanks

**Comment by hfang-bristol** [comment]
Please check your input. It should be a vector of gene symbols for the function xEnricherGenes. If you have a vector of GeneID you can get converted gene symbols using the function xGeneID2Symbol. Clearly what you have input does not meet the requirements.
